src/dra/42398.c

```diff
@@ -2750,7 +2750,7 @@ void func_800FAC0C(s32 context) { D_80137638[context].unk0 = 2; }
 
 void func_800FAC30(void) {
     D_80137844[0] = 0;
-    D_80137848 = 0;
+    D_80137848[0] = 0;
```

Comment:
I think that `D_80137848` is not an array but a pointer, therefore it can be simplified as `*D_80137848 = 0`

Reply:
Unless I'm missing something, it seems like GCC doesn't consider `D_80137848[0]` to be equivalent to `*D_80137848`.

Using this function as an example:
```
extern s32 D_80137848[];

void func_800FAC30(void) {
    D_80137844[0] = 0;
    D_80137848[0] = 0;
}

TARGET                                                   CURRENT (0)                                
18898:    lui     at,%hi(D_80137844)                     18898:    lui     at,%hi(D_80137844)       
1889c:    sw      zero,%lo(D_80137844)(at)               1889c:    sw      zero,%lo(D_80137844)(at) 
188a0:    lui     at,%hi(D_80137848)                     188a0:    lui     at,%hi(D_80137848)       
188a4:    sw      zero,%lo(D_80137848)(at)               188a4:    sw      zero,%lo(D_80137848)(at) 
188a8:    jr      ra                                     188a8:    jr      ra                       
188ac:    nop                                            188ac:    nop                              
1
```

```
extern s32 *D_80137848;

void func_800FAC30(void) {
    D_80137844[0] = 0;
    *D_80137848 = 0;
}

TARGET                                                   CURRENT (795)                              
18898:    lui     at,%hi(D_80137844)              r      1889c:    lui     v0,%hi(D_80137848)       
1889c:    sw      zero,%lo(D_80137844)(at)        |      188a0:    lw      v0,%lo(D_80137848)(v0)   
188a0:    lui     at,%hi(D_80137848)              i      188a4:    lui     at,%hi(D_80137844)       
188a4:    sw      zero,%lo(D_80137848)(at)        i      188a8:    sw      zero,%lo(D_80137844)(at) 
188a8:    jr      ra                                     188ac:    jr      ra                       
188ac:    nop                                     |      188b0:    sw      zero,0(v0)               
```